### PR TITLE
setup e2e tests in github actions

### PR DESCRIPTION
**Pull request by boddumanohar**
**What type of PR is this?**
This PR adds setting up e2e testing in GitHub action using [kind](https://kind.sigs.k8s.io/docs/user/quick-start/). Kind helps us do e2e tests on various Kubernetes versions and it's also pretty lightweight.

**What this PR does / why we need it**:
helps in our efforts to move tests to Github actions.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [X] adds unit tests
- [X] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: test/utils/deploy-kind.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#!/bin/bash
+
+KUBERNETES_VERSION=v1.18.8
+KUBECONFIG=$HOME/.kube/config
+
+# Setup and download kind
+curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.9.0/kind-linux-amd64 && chmod +x ./kind && sudo mv ./kind /usr/local/bin/kind
+curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.18.8/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+
+sudo kind create cluster --image kindest/node:$KUBERNETES_VERSION
```
Comment (andyzhangx) [inline test/utils/deploy-kind.sh:24]: does kind supports windows cluster?

Comment (boddumanohar) [reply test/utils/deploy-kind.sh:24]: No, Kind currently only supports Linux nodes.

Comment (boddumanohar) [reply test/utils/deploy-kind.sh:24]: https://kubernetes.slack.com/archives/CEKK1KTN2/p1601710766100300

File: test/utils/deploy-kind.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# limitations under the License.
+
+#!/bin/bash
+
+KUBERNETES_VERSION=v1.18.8
+KUBECONFIG=$HOME/.kube/config
+
+# Setup and download kind
+curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.9.0/kind-linux-amd64 && chmod +x ./kind && sudo mv ./kind /usr/local/bin/kind
+curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.18.8/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
```
Comment (andyzhangx) [inline test/utils/deploy-kind.sh:22]: `v1.18.8` --> `$KUBERNETES_VERSION`

File: test/utils/deploy-kind.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+sudo kind create cluster --image kindest/node:$KUBERNETES_VERSION
+mkdir -p $HOME/.kube
+sudo chown -R $USER: $HOME/.kube/
+sudo kind get kubeconfig > $KUBECONFIG
+
+# setup samba server and deploy SMB CSI driver
```
Comment (andyzhangx) [inline test/utils/deploy-kind.sh:29]: `echo "setup samba server and deploy SMB CSI driver ..."`

File: .github/workflows/ubuntu.yml [hunk 4]
```diff
@@ -46,6 +46,12 @@ jobs:
         make
         make sanity-test
 
+    - name: e2e tests
```
Comment (andyzhangx) [inline .github/workflows/ubuntu.yml:49]: could you move this to a standalone yml file, could be `ubuntu-e2e.yml`

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**